receiver/kubeletstatsreceiver/kubelet/client.go

```diff
@@ -64,44 +54,90 @@ func newTLSClient(endpoint string, cfg *ClientConfig, logger *zap.Logger) (Clien
 	if err != nil {
 		return nil, err
 	}
-	return defaultTLSClient(endpoint, cfg.InsecureSkipVerify, rootCAs, clientCert, logger), nil
+	return defaultTLSClient(
+		endpoint,
+		cfg.InsecureSkipVerify,
+		rootCAs,
+		[]tls.Certificate{clientCert},
+		nil,
+		logger,
+	), nil
 }
 
-func defaultTLSClient(endpoint string, insecureSkipVerify bool, rootCAs *x509.CertPool, clientCert tls.Certificate, logger *zap.Logger) *tlsClient {
+func newServiceAccountClient(endpoint string, caCertPath string, tokenPath string, logger *zap.Logger) (*clientImpl, error) {
+	rootCAs, err := systemCertPoolPlusPath(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+	tok, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		return nil, err
+	}
+	tr := defaultTransport()
+	tr.TLSClientConfig = &tls.Config{
+		RootCAs: rootCAs,
+	}
+	return defaultTLSClient(endpoint, true, rootCAs, nil, tok, logger), nil
+}
+
+func defaultTLSClient(
+	endpoint string,
+	insecureSkipVerify bool,
+	rootCAs *x509.CertPool,
+	certificates []tls.Certificate,
+	tok []byte,
+	logger *zap.Logger,
+) *clientImpl {
 	tr := defaultTransport()
 	tr.TLSClientConfig = &tls.Config{
 		RootCAs:            rootCAs,
-		Certificates:       []tls.Certificate{clientCert},
+		Certificates:       certificates,
 		InsecureSkipVerify: insecureSkipVerify,
 	}
-	return &tlsClient{
+	if endpoint == "" {
+		endpoint = defaultEndpoint(logger)
```

Comment:
I would put a log entry here like (maybe warning?): kubelet endpoint is not provided, using hostname "..." by default.

I think usually it's not going to work so it would be good to let user know about it, in case if they just forgot to set the endpoint config.

Reply:
This functionality is borrowed from the Smart Agent and I agree it's not obvious. If you set hostNetwork to true in the pod spec, the pod has access to the node's loopback device. I have added a comment to explain, have added a log warning, and updated the readme.

Reply:
Cool. Thank you!